### PR TITLE
Default saved search sorting to date [PD-1014]

### PR DESCRIPTION
**Pull request by almathew**
All tests pass

### CODE REVIEW

File: mysearches/migrations/0007_partner_saved_search_default_sort.py [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        # Default sort_by set to 'Date' for Partner Saved Searches, see PD-912
+        orm.SavedSearch.objects.filter(partnersavedsearch__isnull=False).update(sort_by='Date')
```
Comment (almathew) [inline mysearches/migrations/0007_partner_saved_search_default_sort.py]: Assuming the fix actually works this time it shouldn't be needed again.

File: mysearches/forms.py [hunk 2]
```diff
@@ -219,6 +219,8 @@ def clean(self):
     def save(self, commit=True):
         self.instance.feed = self.cleaned_data.get('feed')
         is_new_or_change = CHANGE if self.instance.pk else ADDITION
+        if not self.instance.pk:
```
Comment (aspidites) [inline mysearches/forms.py]: not having a pk implies a new form and thus not user specified?

File: mysearches/forms.py [hunk 3]
```diff
@@ -232,7 +234,6 @@ def save(self, commit=True):
             }
             Invitation(**invite_args).save()
             # Default sort_by for new Partner Saved Searches, see PD-912
```
Comment (aspidites) [inline mysearches/forms.py]: If you're removing the code, the comment can probably be removed or updated and moved to your changes above, right?

File: mysearches/forms.py (hunk 2 above)
Comment (almathew) [inline mysearches/forms.py]: Implies new form and therefore default should be date. I don't really care about the user.

File: mysearches/forms.py (hunk 3 above)
Comment (almathew) [inline mysearches/forms.py]: I've updated my comment. The datamigration is purely historical, so I'm just going to leave it as-is.

**Pull request merged**